lib/afmotion/http_client.rb

```diff
@@ -62,6 +65,12 @@ def request_serializer(serializer)
       else
         @operation_manager.requestSerializer = serializer
       end
+      reinit_options
+    end
+
+    def reinit_options
+      @headers.each{ |h,v| header(h, v) } if !@headers.nil? && !@headers.empty?
```

Comment:
`if !@headers.nil? && !@headers.empty?` has a lot of logic to mentally unpack

`unless @headers.nil? || @headers.empty?` would probably read more clearly